src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/SdkCollection.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Build.Construction;
+
+namespace Microsoft.DotNet.UpgradeAssistant.MSBuild
+{
+    public class SdkCollection : ICollection<string>
+    {
+        private readonly ProjectRootElement _projectRoot;
+
+        public SdkCollection(ProjectRootElement projectRoot)
+        {
+            _projectRoot = projectRoot;
+        }
+
+        public HashSet<string> GetSdks() => _projectRoot.Sdk.Split(';').ToHashSet(StringComparer.OrdinalIgnoreCase);
+
+        public ICollection<string> GetSdkLists() => _projectRoot.Sdk.Split(';').ToList();
+
+        public bool Contains(string item)
+        {
+            return GetSdks().Contains(item, StringComparer.OrdinalIgnoreCase);
+        }
+
+        public void Add(string item)
+        {
+            if (!Contains(item))
+            {
+                _projectRoot.Sdk = string.Concat(_projectRoot.Sdk, ";", item);
+            }
+        }
+
+        public void Clear()
+        {
+            _projectRoot.Sdk = string.Empty;
+        }
+
+        public int Count
+        {
+            get
+            {
+                return GetSdks().Count;
+            }
+        }
+
+        public bool IsReadOnly
+        {
+            get { return false; }
+        }
+
+        public bool Remove(string item)
+        {
+            if (Contains(item))
```

Comment:
This is going to create the hashset and then a list. Can it be simplified?

Reply:
I can just use the Hashset to remove but it wont retain the order which might be Ok. But is there a better way of doing this without getting into string manipulation?

Reply:
Tried with LINQ

Reply:
This doesn't work - you're still creating a copy twice, but now it's the hashset twice. The hashset doesn't maintain the order (I'm assuming that matters). If you're going to us linq, do it on the GetSdksList